packages/oats-runtime/src/make.ts

```diff
@@ -533,6 +444,113 @@ export function makeNullable(maker: any) {
   };
 }
 
+function isScalar(type: Type): boolean {
+  if (type.type === 'named') {
+    return isScalar(type.reference.definition);
+  }
+  return ['array', 'object', 'union', 'intersection', 'unknown'].indexOf(type.type) < -1;
+}
+
+function isEnum(type: Type) {
+  if (type.type === 'null') {
+    return true;
+  }
+  if (
+    type.type === 'string' ||
+    type.type === 'boolean' ||
+    type.type === 'number' ||
+    type.type === 'integer'
+  ) {
+    return !!type.enum;
+  }
+  return false;
+}
+
+function priority(v: ObjectType['properties'][0]) {
+  // check scalars first to avoid constructing trees unnecessarily
+  if (isScalar(v.value)) {
+    if (isEnum(v.value)) {
+      // high chance this is a union type tag
+      if ((v.value as any).enum?.length === 1) {
+        return 4;
+      }
+      // enum matches less things than non enum
+      return 3;
+    }
+    if (v.value.type === 'string') {
+      // a pattern matches less things than a non pattern so lets try this first
```

Comment:
true but it might also be slower, specially if we are compiling a new regex every time

Reply:
all the makers are cached on first run so we *should* not. I'm willing to be edumacated though

Reply:
If we are *probably* caching the regexes, and doing this just to try to squeeze some performance gains out of making, I think this is fine 👍